src/rag/embeddings.py

```
@@ -1,58 +1,132 @@
+            # Fallback if already a list-like
+            if isinstance(emb, list):
+                return emb
+        elif isinstance(response, dict):
+            emb = response.get("embedding")
+            if isinstance(emb, dict) and "values" in emb:
+                return list(emb["values"])
+            if isinstance(emb, list):
+                return emb
+
+        logger.warning(f"Unexpected embedding response structure: {type(response)}")
+        return None
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)